src/DateDuration.ts

```diff
@@ -0,0 +1,97 @@
+import { parse } from "tinyduration";
+import { LocalDateUnit } from "./LocalDate";
+
+export type DateDurationUnit = LocalDateUnit;
+
+export class DateDuration {
+  public readonly value: ValidDateDurationObject;
+
+  constructor(readonly input: DateDurationObject) {
+    if (!validateDurationObject(input)) {
+      throw new Error("Provided duration object is not valid");
+    }
+    this.value = input;
+  }
+
+  static from(value: string | DateDuration | DateDurationObject): DateDuration {
+    if (value instanceof DateDuration) {
+      return value;
+    }
+
+    if (typeof value === "string") {
+      return DateDuration.fromString(value);
+    }
+
+    return DateDuration.fromObject(value);
+  }
+
+  static fromString(string: string): DateDuration {
+    const parsedDuration = parse(string);
+    return DateDuration.fromObject({
+      years: parsedDuration.years,
+      months: parsedDuration.months,
+      weeks: parsedDuration.weeks,
+      days: parsedDuration.days,
+    });
+  }
+
+  static fromObject({
+    years = 0,
+    months = 0,
+    weeks = 0,
+    days = 0,
+  }: DateDurationObject): DateDuration {
+    return new DateDuration({ years, months, weeks, days });
+  }
+
+  format(): string {
+    const yearText = formatUnit(this.value.years, "year", "years");
+    const monthText = formatUnit(this.value.months, "month", "months");
+    const weekText = formatUnit(this.value.weeks, "week", "weeks");
+    const dayText = formatUnit(this.value.days, "day", "days");
+    return `${yearText}${monthText}${weekText}${dayText}`.trim();
+  }
```

Comment:
NIT: the spaces being built into the formatUnit is a bit unexpected. Was trying to work out what would needed to be trimmed. I think something like this might be better.

```suggestion
  format(config?: { seporator?: string}): string {
    const yearText = formatUnit(this.value.years, "year", "years");
    const monthText = formatUnit(this.value.months, "month", "months");
    const weekText = formatUnit(this.value.weeks, "week", "weeks");
    const dayText = formatUnit(this.value.days, "day", "days");
    return [yearText, monthText, weekText, dayText].filter(Boolean).join(config?.separator ?? " ")
  }
```

maybe even better would be to have format unit print out `0 days` as the empty string may also be unexpected and have this function filter out 0's before continuing (could also be configured if you want 0's to be filtered...)

As it's only used here, you could just assume there's enough context.